src/js/checklist.js

```diff
@@ -58,10 +58,44 @@ function populateChecklistFromLocalStorage(checkboxSelector) {
 
 function processChecklist() {
 	var checkboxSelector = '.c-checklist__checkbox input[type="checkbox"]';
-
 	populateChecklistFromLocalStorage(checkboxSelector);
 	processChecklistClick(checkboxSelector);
 }
 
 openLinkedCheckListItem();
 processChecklist();
+
+function registerToggleButton(buttonEl, parentEl) {
+	if (buttonEl && parentEl) {
+		var details = parentEl.querySelectorAll("details");
+		buttonEl.addEventListener("click", function (event) {
+			details.forEach(function (item) {
+				if (item.hasAttribute("open") === true) {
+					item.removeAttribute("open");
+				} else {
+					item.setAttribute("open", true);
+				}
+			});
+		});
+	}
+}
+
+function renderToggleAll() {
+	var toggleAllButton = document.querySelector("#toggle-all");
+	registerToggleButton(toggleAllButton, document);
+}
+
+renderToggleAll();
+
+function renderToggleCategories(){
```

Comment:
🔍 You can probably write a query selector that gets both `#toggle-all` _and_ all the other buttons, which then means you won't need a dedicated function for `renderToggleAll`.

Reply:
Done!